metaflow/task.py

```diff
@@ -188,10 +197,10 @@ def lineage():
                     if join_type == "foreach":
                         top = i["_foreach_stack"][-1]
                         bottom = i["_foreach_stack"][:-1]
-                        # the topmost indices in the stack are all
-                        # different naturally, so ignore them in the
+                        # the topmost indices and values in the stack are
+                        # all different naturally, so ignore them in the
                         # assertion
-                        yield bottom + [top._replace(index=0)]
+                        yield bottom + [top._replace(index=0, value=0)]
```

Comment:
shouldn't it be value=None?

Reply:
we dont use index/value in this `lineage()`, so it should be the same giving them whatever value. (keeping it 0 to be consistent with index).